open-ce/inputs.py

```diff
@@ -230,18 +240,24 @@ def parse_args(parser, arg_strings=None):
     '''
     Parses input arguments and handles more complex defaults.
     - conda_build_config: If not passed in the default is with the env file,
-                          if is passed in, otherwise it is  expected ot be in
+                          if is passed in, otherwise it is  expected to be in
                           the local path.
     '''
     args = parser.parse_args(arg_strings)
-    if "conda_build_config" in vars(args).keys() and args.conda_build_config is None:
-        if "env_config_file" in vars(args).keys() and args.env_config_file:
-            args.conda_build_config = os.path.abspath(os.path.join(os.path.dirname(args.env_config_file[0]),
-                                                                                   utils.CONDA_BUILD_CONFIG_FILE))
+
+    if "conda_build_config" in vars(args).keys():
+        if args.conda_build_config is None:
+            if "env_config_file" in vars(args).keys() and args.env_config_file:
+                args.conda_build_config = os.path.join(os.path.dirname(args.env_config_file[0]),
+                                                       utils.CONDA_BUILD_CONFIG_FILE)
+            else:
+                args.conda_build_config = utils.DEFAULT_CONDA_BUILD_CONFIG
+
+        if utils.is_url(args.conda_build_config):
+            args.conda_build_config, _ = urllib.request.urlretrieve(args.conda_build_config)
```

Comment:
This should probably be in a try block, but maybe this can be handled as part of #310.

Reply:
I mentioned above that the error you get is pretty useful from our current try blocks:
```
$ open-ce/open-ce build env https://raw.githubusercontent.com/bethune-bryant/open-ce-environments/test_branch/envs/test-env.yam --skip_build_packages --repository_folder /tmp/bryant/test
[OPEN-CE-ERROR-10] Error creating Build Tree
[OPEN-CE-ERROR-0] Unexpected Error: Error in https://raw.githubusercontent.com/bethune-bryant/open-ce-environments/test_branch/envs/test-env.yam:
  HTTP Error 404: Not Found
```

Do you think we should add an explicit error?

Reply:
I think I should do this as part of this issue. I'm looking into it.

Reply:
Okay, this is what a failure looks like now:
```
$ open-ce/open-ce build env https://raw.githubusercontent.com/bethune-bryant/open-ce-environments/test_branch/envs/test-env.yaml --skip_build_packages --repository_folder /tmp/bryant/test --conda_build_config https://doesnotexisttest.com/cond_build_config.yaml
[OPEN-CE-ERROR-24] Failed to download https://doesnotexisttest.com/cond_build_config.yaml with error:
<urlopen error [Errno -2] Name or service not known>
```